src/systemtest/resources/noDateRange/expected/summary.json

```diff
@@ -35,6 +35,13 @@
       "repoName": "testrepo-Delta",
       "branch": "nonExistentBranch",
       "displayName": "reposense_testrepo-Delta_nonExistentBranch"
+    },
+    {
+      "location": "https://github.com/reposense/testrepo-Delta.git",
+      "organization": "reposense",
+      "repoName": "testrepo-Delta",
+      "branch": "HEAD",
```

Comment:
Would prefer that it retrieves the name of the current working branch.

Reply:
Not sure how to implement this as my default branch is HEAD

Reply:
> Not sure how to implement this as my default branch is HEAD

You can look for ways to extract the name of current branch from the git repository.

Reply:
Yes, however there are also drawbacks in some scenarios
1. Using `HEAD` as the branch name in report may not be intuitive (low)
2. Simply using `HEAD` may cause some obstructions to implementing future features and issues. E.g. #337 (medium-high)

Since, you can extract the branch name using the combination of `git status` and `StringsUtil.filterText`, let's resolve it within this PR itself.

Reply:
>  However, the display name is used when cloning the repository, and display name requires a branch parameter, which is HEAD by default i.e. testrepo-DELTA_HEAD. Branch name cannot be obtained using git status until after the git repository has been cloned, by which time the directories has already been created.

I see. I think that is a legacy problem, we can probably delay the creation of the directories till right before the repository files are generated if that is a problem.

Reply:
If you mean the `repos` directory, however, that is just a temporary storage for the clone repository. We can use a different or constant name since its deleted right before we clone the next repository.